pkgs/top-level/config.nix

```diff
@@ -195,8 +195,44 @@ let
       '';
     };
 
+    allowlistedLicenses = mkOption {
+      description = ''
+        Allow licenses that are specifically acceptable. `allowlistedLicenses` only applies to unfree licenses unless
+        `allowUnfree` is enabled. It is not a generic allowlist for all types of licenses.
+      '';
+      default = [ ];
+      type = types.listOf (types.attrsOf types.anything);
+      example = literalExpression ''
+        with lib.licenses; [
+          amd
+          wtfpl
+        ]
+      '';
+    };
```

Comment:
Rhetorical question: how does one access `lib.licenses` on the first `import nixpkgs` instantiation?

Reply:
I know you said rhetorical... but I only recently found out that `config` can also be a function which accepts an attribute set containing only `pkgs`:

https://github.com/NixOS/nixpkgs/blob/31c1c19e4828ef391f8d80310031de51306b26de/pkgs/top-level/default.nix#L114-L117

So... I guess you *can* use `pkgs` (and subsequently, `lib`) in `config`.